### PR TITLE
fix: Supports using decimal in cluster+adv_cluster advanced_configuration `oplog_min_retention_hours`

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Supports using decimal in cluster+adv_cluster advanced_configuration `oplog_min_retention_hours`

See also #2594

Link to any related issue(s): CLOUDP-273994

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/advancedcluster/model_advanced_cluster.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func SchemaAdvancedConfigDS() *schema.Schema {
 					Computed: true,
 				},
 				"oplog_min_retention_hours": {
-					Type:     schema.TypeInt,
+					Type:     schema.TypeFloat,
```
Comment (lantoli) [inline internal/service/advancedcluster/model_advanced_cluster.go:107]: have you ckecked if there are plan changes for existing clusters?

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go [hunk 2]
```diff
@@ -154,12 +154,18 @@ func TestMigAdvancedCluster_partialAdvancedConf(t *testing.T) {
 	var (
 		projectID   = acc.ProjectIDExecution(t)
 		clusterName = acc.RandomClusterName()
-		extraArgs   = `
+		// necessary to test oplog_min_retention_hours
```
Comment (lantoli) [inline internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:157]: great these comments about WHY  👏

**Review by lantoli — APPROVED** [review]

File: .changelog/2604.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+```release-note:bug
+resource/mongodbatlas_advanced_cluster: Supports using decimal in advanced_configuration `oplog_min_retention_hours`
+```
+
+```release-note:bug
+resource/mongodbatlas_cluster: Supports using decimal in advanced_configuration `oplog_min_retention_hours`
```
Comment (lantoli) [inline .changelog/2604.txt:6]: I can't see any change in cluster, only adv_cluster

File: internal/service/advancedcluster/model_advanced_cluster.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/service/advancedcluster/model_advanced_cluster.go:107]: Yes, I did a manual test. I also updated the migration test to include this.

File: .changelog/2604.txt (hunk 3 above)
Comment (EspenAlbert) [reply .changelog/2604.txt:6]: cluster references the advanced config [here](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-273994_oplog_min_retention_hours_allow_float/internal/service/cluster/resource_cluster.go#L312)

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go [hunk 4]
```diff
@@ -217,11 +224,12 @@ func TestMigAdvancedCluster_partialAdvancedConf(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "bi_connector_config.0.read_preference", "secondary"),
 				),
 			},
+			mig.TestStepCheckEmptyPlan(configUpdated),
```
Comment (AgustinBettati) [inline internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:227]: could we move this check above the step that actually applies the updated config? Essentially ensure that without applying the new config, an empty plan is encountered.

**Review by AgustinBettati — COMMENTED** [review]
small consideration in migration test to ensure no breaking changes are faced

File: .changelog/2604.txt (hunk 3 above)
Comment (lantoli) [reply .changelog/2604.txt:6]: oh, ok, thanks.  could we update cluster tests then (or create a new one) ?

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go (hunk 4 above)
Comment (AgustinBettati) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:227]: my bad, I believe the existing test covers it correctly: 

- first step defines the config with int using latest provider version (schema with int)
- second step continues using int but uses local built provider that has float schema, and verifies empty plan before applying

I will let you confirm but I believe we are good

**Review by carriecwk — APPROVED** [review]
LGTM

File: internal/service/advancedcluster/resource_advanced_cluster_migration_test.go (hunk 4 above)
Comment (EspenAlbert) [reply internal/service/advancedcluster/resource_advanced_cluster_migration_test.go:227]: Yes, that is correct 👍
Added the `mig.TestStepCheckEmptyPlan(configUpdated)` L227 to check the updated config doesn't have any plan changes (step 4, it was missing). However, the relevant check for this PR happens in the first and second step.

File: .changelog/2604.txt (hunk 3 above)
Comment (EspenAlbert) [reply .changelog/2604.txt:6]: I don't think it is necessary. Both use SDK v2.

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: .changelog/2604.txt (hunk 3 above)
Comment (EspenAlbert) [reply .changelog/2604.txt:6]: *Turns out this test exposed another bug. Good to test 😅

Comment (EspenAlbert) [reply .changelog/2604.txt:6]: Will continue to work on that here
https://jira.mongodb.org/browse/CLOUDP-274185

**Pull request merged**